### ISSUE TITLE
Add capability to decrypt contest data with guardian shares

### ISSUE THREAD

**Issue by JohnLCaron** [danwallach/electionguard-kotlin-multiplatform#160]
See sec 3.5.3

**Comment by JohnLCaron** [comment]
Im having trouble understanding the notation for partial decryption of contest data, section 3.5.3. Im guessing you could do this a lot faster. If you can add the basic methods, I can add it into the decryption workflow.

Im assigning to you, but if you prefer, you can give me the basic idea, and ill i slog through the code.

**Comment by danwallach** [comment]
Is this one of the things that's going to change at some point soon or is this stable?

This is fine for my todo list.

**Comment by JohnLCaron** [comment]
Seems likely the crypto part is stable.

**Comment by danwallach** [comment]
Happy to work on this. 

I haven't looked at all the crypto shares code before. I see that we've got `ElGamalCiphertext.decryptWithShares`, which multiplies together all the shares, yielding the magic value to divide with the ciphertext. What about when we only have a fraction of the shares? Do we have that implemented elsewhere?

**Comment by JohnLCaron** [comment]
Hmm, not sure why we have ElGamalCiphertext.decryptWithShares(), except maybe I added it in the early days for testing.

It appears from section 3.5.4 "Decryption of contest data" that we need a service from the decrypting trustees where its given an ElementModP (C0) and it returns  m_i = C0^t_i mod p. Then you multiply all those together to get β = C0^s. So I can do that part. 

Forgetting the proof for now, I dont understand how you recover the byte array D, from the math in eq 75 - 77. If @danwallach could do that, then I think we'd be mostly done. 

So we need HashedElGamalCiphertext.decryptWithBeta(beta: ElementModP): ByteArray

Pretty sure I can add the proof, since it involves another orchestration from the trustees. I just went through all this with the ElGamal decryption, in 3.5.2 and 3.5.3, using classes Decryptor and TallyDecryptor. Maybe I can reuse or generalize that.
